src/excmds.ts

```diff
@@ -686,6 +686,21 @@ export async function tabmove(index = "0") {
     let newindex: number
     if (index.startsWith("+") || index.startsWith("-")) {
         newindex = Math.max(0, Number(index) + aTab.index)
+        if (newindex === aTab.index) return
+        let candidates = await browser.tabs.query({windowId: aTab.windowId})
+        newindex = Math.min(newindex, candidates.length)
+        candidates.sort((a,b) => {
+            return a.index < b.index ? -1 : 1
+        })
+        candidates = newindex < aTab.index
+            ? candidates.slice(newindex, aTab.index)
+            : candidates.slice(aTab.index + 1, newindex + 1).reverse()
+        for (const candidate of candidates) {
+            if (aTab.pinned === candidate.pinned) {
+                newindex = candidate.index
+                break
+            }
+        }
     } else newindex = Number(index) - 1
```

Comment:
In Vimperator, tabs move as far as they can in their zone. So `:tabmove 10` on a pinned tab will probably move it to the end of the pinned section and `:tabmove 0` on a non-pinned tab will move it to the start of the non-pinned section.

Could put your above code in a new function and call that?

Could just query by pinned tabs in the current window, find the max index and then adjust the start position accordingly?

Reply:
Oh, good catch. I probably should've tried that.